src/backend/distributed/utils/node_metadata.c

```diff
@@ -1088,38 +1240,38 @@ AddNodeMetadata(char *nodeName, int32 nodePort,
 		SendCommandToWorkers(WORKERS_WITH_METADATA, nodeInsertCommand);
 	}
 
-	return nextNodeIdInt;
+	return workerNode->nodeId;
 }
 
 
 /*
- * SetNodeState function sets the isactive column of the specified worker in
- * pg_dist_node to isActive.
+ * SetIsDataNode function sets the column with the specified index on the
+ * worker in pg_dist_node.
+ * It returns the new worker node after the modification.
  */
-static void
-SetNodeState(char *nodeName, int32 nodePort, bool isActive)
+static WorkerNode *
+SetWorkerColumn(WorkerNode *workerNode, int columnIndex, Datum value, char *updateCommand)
```

Comment:
I read the function signature and comment several times, but couldn't understand what this does. It seems more like a hack to save a few lines of code rather than a useful abstraction that helps make the code more maintainable.

Maybe something like: 
```c
static void
UpdateSingleNodeProperty(int nodeId, NodeProperty property, Datum newValue)
{
    ... 
    switch (property)
    {
        case NODE_PROPERTY_IS_DATA_NODE:
        {
            columnIndex = Anum_pg_dist_node_isdatanode;
            metadataSyncCommand = IsDataNodeUpdateCommand(nodeId, DatumGetObjectId(newValue));
            break;
        }
        ....
}

```